packages/workbox-background-sync/src/lib/request-manager.js

```diff
@@ -42,49 +42,60 @@ class RequestManager {
   }
 
   /**
+   * function to play one single request
+   * given its hash
+   * @param {String=} hash
+   * @return {Promise}
+   *
+   * @memberOf RequestManager
+   * @private
+   */
+  async replayRequest(hash) {
+    try {
+      const reqData = await this._queue.getRequestFromQueue({hash});
+      const request = await getFetchableRequest({
+        idbRequestObject: reqData.request});
+      const response = await fetch(request);
+      if(!response.ok) {
+        return Promise.reject(response.status);
```

Comment:
fixed this, moved this to parent function to catch all kind of rejections at one place